test/e2e/runner/setup.go

```diff
@@ -150,9 +150,13 @@ func MakeGenesis(testnet *e2e.Testnet) (types.GenesisDoc, error) {
 	if testnet.PbtsUpdateHeight == -1 {
 		genesis.ConsensusParams.Feature.PbtsEnableHeight = testnet.PbtsEnableHeight
 	}
-	for validator, power := range testnet.Validators {
+	for valName, power := range *testnet.Manifest.ValidatorsMap {
+		validator := testnet.LookupNode(valName)
+		if validator == nil {
+			return types.GenesisDoc{}, fmt.Errorf("unknown validator %q for genesis doc", valName)
```

Comment:
Should never happen, right?

Reply:
Yes, but as suggested by @alesforz and @andynog , it's cleaner code if we check for `nil`.